src/main/java/com/mixpanel/android/viewcrawler/EditProtocol.java

```diff
@@ -90,35 +90,53 @@ public ViewVisitor readEdit(JSONObject source) throws BadInstructionsException {
+                } catch (final ClassNotFoundException e) {
+                    throw new BadInstructionsException("Can't find class for visit path: " + targetClassName, e);
+                }
 
-            final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
-            final JSONArray argsAndTypes = source.getJSONArray("args");
-            final Object[] methodArgs = new Object[argsAndTypes.length()];
-            for (int i = 0; i < argsAndTypes.length(); i++) {
-                final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
-                final Object jsonArg = argPlusType.get(0);
-                final String argType = argPlusType.getString(1);
-                methodArgs[i] = convertArgument(jsonArg, argType);
-            }
+                final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
+                final JSONArray argsAndTypes = source.getJSONArray("args");
+                final Object[] methodArgs = new Object[argsAndTypes.length()];
+                for (int i = 0; i < argsAndTypes.length(); i++) {
+                    final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
+                    final Object jsonArg = argPlusType.get(0);
+                    final String argType = argPlusType.getString(1);
+                    methodArgs[i] = convertArgument(jsonArg, argType);
+                }
 
-            final Caller mutator = prop.makeMutator(methodArgs);
-            if (null == mutator) {
-                throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
-            }
+                final PropertyCaller mutator = prop.makeMutator(methodArgs);
+                if (null == mutator) {
+                    throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
+                }
+
+                return new ViewVisitor.PropertySetVisitor(path, mutator, prop.accessor);
+            } else if (source.has("is_layout")) {
+                final JSONArray args = source.getJSONArray("args");
+                JSONObject layout_info = args.optJSONObject(0);
+                JSONObject params = new JSONObject();
+                params.put("verb", layout_info.getInt("rule_id"));
+                if (layout_info.getString("operation").equals("remove")) {
+                    params.put("anchor", 0);
+                } else if (layout_info.has("anchor_id")) {
+                    params.put("anchor", layout_info.getInt("anchor_id"));
+                } else {
+                    params.put("anchor", RelativeLayout.TRUE);
+                }
 
-            return new ViewVisitor.PropertySetVisitor(path, mutator, prop.accessor);
+                return new ViewVisitor.LayoutSetVisitor(path, params);
```

Comment:
Consider renaming LayoutSetVisitor to LayoutChangeVisitor- "Set" can mean "assign", or "an unordered collection"